src/Type/Php/ArrayFilterFunctionReturnTypeReturnTypeExtension.php

```diff
@@ -54,9 +55,13 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 		}
 
 		$arrayArgType = $scope->getType($arrayArg);
+		$arrayArgType->isArray();
```

Comment:
This is a no-op.

Reply:
The right solution is:

```php
$arrays = TypeUtils::toBenevolentUnion($arrayArgType)->getArrays();
$arrayArgType = TypeCombinator::union(...$arrays);
```

Reply:
Ah, whoops that was a debugging line I thought I had deleted. Thanks for the direction, I will try it out.

Reply:
Or maybe just would work too:

```php
$arrayArgType = TypeUtils::toBenevolentUnion($arrayArgType);
```

Reply:
That worked!

Reply:
And please listen to the bot 😊 https://github.com/phpstan/phpstan-src/pull/2794#issuecomment-1830755596

Reply:
I'll rebase off of 1.10.x and change the PR, I was waiting for "this is a horrible wrong approach" or not. Rebasing now

Reply:
Okay, rebased off of 1.10.x and changed the PR base